gubled/gubled_test.go

```diff
@@ -140,7 +140,13 @@ func TestGcmOnlyStartedIfEnabled(t *testing.T) {
 	defer mockCtrl.Finish()
 
 	routerMock := NewMockPubSubSource(mockCtrl)
+
+	amMock := NewMockAccessManager(mockCtrl)
+	msMock := NewMockMessageStore(mockCtrl)
+
 	routerMock.EXPECT().KVStore().Return(store.NewMemoryKVStore(), nil)
+	routerMock.EXPECT().AccessManager().Return(amMock, nil).MaxTimes(2)
+	routerMock.EXPECT().MessageStore().Return(msMock, nil).MaxTimes(2)
```

Comment:
Maybe it would be better to have a had written struct based implementation of the router interface for testing,
then I could write something like
routerMock := routerMock{MessageStore: NewMockMessageStore(mockCtrl), ...}